app/src/main/java/com/zulip/android/ZulipApp.java

```diff
@@ -175,14 +181,48 @@ public ZulipServices getZulipServices() {
                             .addInterceptor(new ZulipInterceptor())
                             .addInterceptor(logging)
                             .build())
-                    .addConverterFactory(GsonConverterFactory.create())
+                    .addConverterFactory(GsonConverterFactory.create(buildGson()))
                     .baseUrl(getServerURI())
                     .build()
                     .create(ZulipServices.class);
         }
         return zulipServices;
     }
 
+    private Gson buildGson() {
+        return new GsonBuilder()
+                .registerTypeAdapter(UserConfigurationResponse.class, new TypeAdapter<UserConfigurationResponse>() {
+                    final Gson gson = new Gson();
+
+                    @Override
+                    public void write(JsonWriter out, UserConfigurationResponse value) throws IOException {
+                        gson.toJson(gson.toJsonTree(value), out);
+                    }
+
+                    @Override
+                    public UserConfigurationResponse read(JsonReader in) throws IOException {
+                        UserConfigurationResponse res = gson.fromJson(in, UserConfigurationResponse.class);
+
+                        RuntimeExceptionDao<Person, Object> personDao = ZulipApp.this.getDao(Person.class);
+                        for (int i = 0; i < res.getRealmUsers().size(); i++) {
+
+                            Person currentPerson = res.getRealmUsers().get(i);
+                            Person foundPerson = null;
+                            try {
+                                foundPerson = personDao.queryBuilder().where().eq(Person.EMAIL_FIELD, currentPerson.getEmail()).queryForFirst();
+                                if(foundPerson != null) {
+                                    currentPerson.setId(foundPerson.getId());
+                                }
+                            } catch (SQLException e) {
+                                e.printStackTrace();
```

Comment:
I realize this may be code that's been copy pasted from elsewhere, but since it's here, as a rule this should print to either `Zlog` or `Log`.